### PR TITLE
feat(admin-api+ui): add storage usage info for application

### PR DESCRIPTION
**Pull request by frdomovic**
# feat(admin-api+ui): add storage usage info for application

## Summary
Added api for fetching storage usage of application. Right now it only returns the size of the wasm file but guess @miraclx can fill out this info better

Fixes # C20-88
Screenshot 2024-06-05 at 17 55 02

## Summary by CodeRabbit

- **New Features**
  - Added functionality to retrieve and display storage information for applications.
  - Updated UI to show storage usage in megabytes on various components.

- **Dependency Updates**
  - Upgraded `@calimero-is-near/calimero-p2p-sdk` from version `"0.0.14"` to `"0.0.15"`.

### CODE REVIEW

File: node-ui/src/api/dataSource/NodeDataSource.ts [hunk 1]
```diff
@@ -63,6 +63,10 @@ interface RootkeyResponse {
   root_keys: ApiRootKey[];
 }
 
+export interface ApplicationStorageResponse {
+  size: string;
```
Comment (MatejVukosav) [inline node-ui/src/api/dataSource/NodeDataSource.ts]: size in which unit?

Comment (frdomovic) [reply node-ui/src/api/dataSource/NodeDataSource.ts]: updated

File: crates/application/src/lib.rs [hunk 2]
```diff
@@ -231,4 +232,29 @@ impl ApplicationManager {
             None
         }
     }
+
+    pub async fn get_application_storage(
+        &self,
+        application_id: &calimero_primitives::application::ApplicationId,
+        version: &semver::Version,
+    ) -> eyre::Result<StorageInfo> {
+        let base_path = format!(
```
Comment (MatejVukosav) [inline crates/application/src/lib.rs]: This is not correct so we should not serve misleading data.

File: crates/primitives/src/application.rs [hunk 3]
```diff
@@ -42,3 +42,10 @@ pub struct Release {
     pub path: String,
     pub hash: String,
 }
+
+
+#[derive(Clone, Serialize, Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+pub struct StorageInfo {
+    pub size_in_mb: f64,
```
Comment (MatejVukosav) [inline crates/primitives/src/application.rs]: Use separate fields, size and unit

File: crates/server-primitives/src/admin.rs [hunk 4]
```diff
@@ -97,3 +97,18 @@ pub struct NodeChallengeMessage {
     pub application_id: String, //optional if challenge is used on admin level
     pub timestamp: i64,
 }
+
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+pub struct GetStorageRequest {
+    pub application_id: calimero_primitives::application::ApplicationId,
```
Comment (MatejVukosav) [inline crates/server-primitives/src/admin.rs]: Storage is per context and not per application

File: crates/server-primitives/src/admin.rs [hunk 5]
```diff
@@ -97,3 +97,18 @@ pub struct NodeChallengeMessage {
...
+    pub version: semver::Version,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+pub struct GetStorageResponse {
+    pub application_id: calimero_primitives::application::ApplicationId,
```
Comment (MatejVukosav) [inline crates/server-primitives/src/admin.rs]: Why do you need appid and version for context storage?

File: crates/server/src/admin/service.rs [hunk 6]
```diff
@@ -72,6 +72,7 @@ pub(crate) fn setup(
         .route("/contexts/:context_id", delete(delete_context_handler))
         .route("/contexts/:context_id", get(get_context_handler))
         .route("/contexts", get(get_contexts_handler))
+        .route("/storage", post(get_application_storage_handler))
```
Comment (MatejVukosav) [inline crates/server/src/admin/service.rs]: contexts/contextId/storage

File: crates/primitives/src/application.rs (hunk 3 above)
Comment (miraclx) [reply crates/primitives/src/application.rs]: Just size in bytes would suffice, consumers can interpret that however they like

File: crates/application/src/lib.rs [hunk 7]
```diff
@@ -231,4 +232,29 @@ impl ApplicationManager {
...
+            "{}/{}/{}/binary.wasm",
+            &self.application_dir, application_id, version
+        );
+        let metadata = match fs::metadata(&base_path) {
+            Ok(metadata) => metadata,
+            Err(_) => return Ok(StorageInfo { size_in_mb: 0.0 }),
+        };
+        let file_size_bytes = metadata.len();
+        let file_size_mb = (file_size_bytes as f64) / (1024.0 * 1024.0);
+        let formatted_size_mb = format!("{:.2}", file_size_mb);
```
Comment (miraclx) [inline crates/application/src/lib.rs]: move this formatting to the consumer

File: crates/primitives/src/application.rs (hunk 3 above)
Comment (frdomovic) [reply crates/primitives/src/application.rs]: updated

File: crates/server/src/admin/service.rs (hunk 6 above)
Comment (frdomovic) [reply crates/server/src/admin/service.rs]: updated

File: crates/primitives/src/application.rs (hunk 3 above)
Comment (MatejVukosav) [reply crates/primitives/src/application.rs]: or basic unit as well.

**Pull request merged**